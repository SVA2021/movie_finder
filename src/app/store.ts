import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import authReducer from '../features/auth/authSlice';
import movieFinderReducer from '../features/movieFinder/movieFinderSlice';

export const store = configureStore({
  reducer: {
    auth: authReducer,
		movieFinder: movieFinderReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
