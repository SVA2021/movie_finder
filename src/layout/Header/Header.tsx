import {AppBar, Box, Grid, Toolbar} from '@mui/material';
import {AccountLabel, Logo, MenuMain, SearchMenu} from '../../components';

export const Header = () => {
  return (
    <Box sx={{flexGrow: 1, bgcolor: 'primary.main'}} data-testid={'header'}>
      <AppBar position='sticky'
        sx={{
          maxWidth: 'xl',
          m: '0 auto',
          zIndex: 900,
          backgroundImage: 'none',
        }}
      >
        <Toolbar sx={{boxShadow: 'none'}}>
          <Grid container justifyContent={'space-between'} alignItems={'center'}>
            <Grid item xs={3} md={1}>
              <MenuMain />
            </Grid>
            <Grid item xs={9} md={5} lg={4}>
              <Logo />
            </Grid>
            <Grid item md={1} lg={4} xl={4}
              sx={{display: {xs: 'none', md: 'block'}}}>
              <SearchMenu />
            </Grid>
            <Grid item md={3} lg={2}
              sx={{display: {xs: 'none', md: 'block'}}}>
              <AccountLabel />
            </Grid>
          </Grid>
        </Toolbar>
      </AppBar>
    </Box>
  )
};