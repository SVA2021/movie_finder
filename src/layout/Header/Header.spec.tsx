import {render, screen} from "@testing-library/react";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import {store} from "../../app/store";
import {Header} from "./Header";

describe('render Header', () => {

  const setUp = () => render(
    <Provider store={store}>
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    </Provider>
  );

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('header')).toMatchSnapshot();
  });

})