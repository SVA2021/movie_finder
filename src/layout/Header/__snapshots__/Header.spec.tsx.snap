// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header snapshot test 1`] = `
<div
  class="MuiBox-root css-7xotxy"
  data-testid="header"
>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-m1z56k-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-13qhygc-MuiToolbar-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-9cyib4-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-1 css-10f59y6-MuiGrid-root"
        >
          <button
            aria-haspopup="true"
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1qbbts5-MuiButtonBase-root-MuiButton-root"
            id="fade-button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 MuiGrid-grid-md-5 MuiGrid-grid-lg-4 css-1d62rrc-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-h1 css-lswgau-MuiTypography-root"
            data-testid="logo"
          >
            Movie
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
              data-testid="LocalMoviesIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 3v2h-2V3H8v2H6V3H4v18h2v-2h2v2h8v-2h2v2h2V3h-2zM8 17H6v-2h2v2zm0-4H6v-2h2v2zm0-4H6V7h2v2zm10 8h-2v-2h2v2zm0-4h-2v-2h2v2zm0-4h-2V7h2v2z"
              />
            </svg>
            Finder
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-1 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4 css-1hub91h-MuiGrid-root"
        >
          <form
            class="MuiBox-root css-hpgf8j"
            data-testid="search-field"
          >
            <div
              class="MuiInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-adornedEnd css-1qmitm0-MuiInputBase-root-MuiInput-root"
            >
              <input
                class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                placeholder="Movies, series, keywords..."
                type="text"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1wgghly-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="submit"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="SearchIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </form>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-3lf7b5-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-h3 css-14at1vf-MuiTypography-root"
            data-testid="styled-text-item"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AccountCircleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
              />
            </svg>
            Гость
          </p>
        </div>
      </div>
    </div>
  </header>
</div>
`;
